src/Microsoft.AspNetCore.Sockets.Client/ServerSentEventsTransport.cs

```diff
@@ -0,0 +1,223 @@
+            var sendTask = SendMessages(sendUrl, _cancellationToken);
+            var openConnectionTask = OpenConnection(_application, sseUrl, _cancellationToken);
+
+            Running = Task.WhenAll(sendTask, openConnectionTask).ContinueWith(t =>
+            {
+                _logger.LogDebug("Transport stopped. Exception: '{0}'", t.Exception?.InnerException);
+
+                _application.Output.TryComplete(t.IsFaulted ? t.Exception.InnerException : null);
+                return t;
+            }).Unwrap();
+
+            return TaskCache.CompletedTask;
+        }
+
+        public async Task OpenConnection(IChannelConnection<SendMessage, Message> application, Uri url, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Starting receive loop");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, url);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/event-stream"));
+            var response = await _httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+
+            var stream = await response.Content.ReadAsStreamAsync();
+
+            var pipelineReader = stream.AsPipelineReader();
+
+            while (true)
+            {
+                var result = await pipelineReader.ReadAsync();
+                var input = result.Buffer;
+                var consumed = input.Start;
+                var examined = input.End;
+
+                try
+                {
+                    if (input.IsEmpty && result.IsCompleted)
+                    {
+                        _logger.LogDebug("Done reading from pipeline");
+                        break;
+                    }
+
+                    var parseResult = ServerSentEventsMessageParser.ParseMessage(input, out consumed, out examined, out var message);
+
+                    switch (parseResult)
+                    {
+                        case ServerSentEventsMessageParser.ParsePhase.Completed:
+                            break;
+                        case ServerSentEventsMessageParser.ParsePhase.Incomplete:
+                            if (result.IsCompleted)
+                            {
+                                throw new FormatException("There was an error parsing");
+                            }
+
+                            continue;
+                        case ServerSentEventsMessageParser.ParsePhase.Error:
+                            throw new FormatException("There was an error parsing");
+                    }
+
+                    _application.Output.TryWrite(message);
+                }
+                finally
+                {
+                    pipelineReader.Advance(consumed, examined);
+                }
+
+            }
+        }
+
+        public async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Starting the send loop");
```

Comment:
We will need to merge this later so that we have one copy for both server sent events and long polling.